evalml/tests/component_tests/test_one_hot_encoder.py

```diff
@@ -550,3 +550,17 @@ def test_ohe_woodwork_custom_overrides_returned_by_components(X_df):
         assert isinstance(transformed, ww.DataTable)
         if logical_type != Categorical:
             assert transformed.logical_types == {0: logical_type}
+
+
+def test_ohe_output_bools():
+    X = ww.DataTable(pd.DataFrame({"bool": [bool(i % 2) for i in range(100)],
+                                   "categorical": ["dog"] * 20 + ["cat"] * 40 + ["fish"] * 40,
+                                   "integers": [i for i in range(100)]}))
+    y = ww.DataColumn(pd.Series([i % 2 for i in range(100)]))
+    ohe = OneHotEncoder()
+    output = ohe.fit_transform(X, y)
+    for name, types in output.types["Logical Type"].items():
+        if name == 'integers':
+            assert str(types) == "Integer"
+        else:
+            assert str(types) == "Boolean"
```

Comment:
Looks great! I love that you made this general and not specific to xgboost.

One check I'd add: assert that the right number of features has been generated. That way, if for some reason the OHE isn't generating any features at all for the bool/category inputs, the test will fail.